src/middlewared/middlewared/plugins/service.py

```diff
@@ -934,6 +934,7 @@ def _started_notify(self, verb, what):
     async def _start_cifs(self, **kwargs):
         await self._service("ix-pre-samba", "start", quiet=True, **kwargs)
         await self._service("samba_server", "start", quiet=True, **kwargs)
+        await self.middleware.call("smb.add_admin_group", "", True)
```

Comment:
add_admin_group can raise ValidationErrors which is not suited for service start.

We should at least catch the exception and re-raise as CallError here.

Reply:
> > Thats up to the developers discretion. Can your team please focus only on what is user-facing? At least for the middleware portion.
> 
> I'll skip repeating what I said earlier, and just add that many of these are github suggestions which can be accepted with a single click, or ignored. I don't understand why that is a problem.

Docs fixes have been made.